src/test/java/de/retest/recheck/testng/RecheckHookTest.java

```diff
@@ -67,6 +74,24 @@ public void usesParametersInTestName() throws Exception {
 
 		hook.run( callBack, testResult );
 
-		verify( testInstance.someSuperSuperClassField ).startTest( combinedName );
+		verify( testInstance.recheck ).startTest( combinedName );
+	}
+
+	@Test
+	public void callCapAfterTestFailed() throws Exception {
+		final AssertionError assertionError = new AssertionError(
+				"You have to catch AssertionErrors generated by capTest in " + RecheckHook.class.getSimpleName() );
+		doThrow( assertionError ).when( testInstance.recheck ).capTest();
+
+		final RecheckHook hook = new RecheckHook();
+
+		try {

```

Comment:
You can leverage AssertJ to make this less verbose, check out our corresponding test in the Jupiter extension: https://github.com/retest/recheck-junit-jupiter-extension/blob/master/src/test/java/de/retest/recheck/junit/jupiter/RecheckExtensionTest.java#L107

Reply:
Simplified using assertj.